### ISSUE TITLE
_MockHttpClientRequest missing abort implementation

### ISSUE THREAD

**Issue by abnerh69** [marcglasberg/network_to_file_image#18]
Hello Marc,  simple and awesome this package.  Now, on Flutter 1.22.0-10.0.pre.39 • channel master •  (not tested in previous versions yet) it appears is missing abort implementation in this class:

class _MockHttpClientRequest extends HttpClientRequest {

// should add ->  @override
// should add ->  void abort([Object exception, StackTrace stackTrace]) {}

}

As you can see here: https://github.com/dart-lang/sdk/issues/22265

**Comment by marcglasberg** [comment]
Thanks! I can't fix this for master. I have to wait until the signature change goes to stable.

**Comment by pmrajani** [comment]
@marcglasberg any update? i faced same issue as  @abnerh69

**Comment by pmrajani** [comment]
/C:/src/flutter/.pub-cache/hosted/pub.dartlang.org/network_to_file_image-2.3.3/lib/network_to_file_image.dart:407:7: Error: The non-abstract class '_MockHttpClientRequest' is missing implementations for these members:
 - HttpClientRequest.abort
Try to either
 - provide an implementation,
 - inherit an implementation from a superclass or mixin,
 - mark the class as abstract, or
 - provide a 'noSuchMethod' implementation.

class _MockHttpClientRequest extends HttpClientRequest {
      ^^^^^^^^^^^^^^^^^^^^^^
org-dartlang-sdk:///third_party/dart/sdk/lib/_http/http.dart:2045:8: Context: 'HttpClientRequest.abort' is defined here.
  void abort([Object? exception, StackTrace? stackTrace]);
       ^^^^^

**Comment by marcglasberg** [comment]
@pmrajani I suggest you downgrade to stable by typing this into your terminal:

```dart
 flutter channel stable
 flutter upgrade
```

I am sorry but I can't fix this for you and break it for every one else on the stable channel.

**Comment by dannycortesv** [comment]
Hi, i'm also having this issue, it is blocking me now, i need to use beta channel to support iOS 14, can't switch to stable

**Comment by marcglasberg** [comment]
Hey guys, I apologize. I haven't paid attention to the fact that `HttpClientRequest`  is from `dart:io`. In this case I don't need to import anything in my `pubspec.yaml`. And since this is just asking for an override it seems I can fix it and it won't break those people on stable.

Please upgrade to version [2.3.4].

@abnerh69 @pmrajani @dannycortesv  Could you please tell me if this issue is fixed?

**Comment by Sam-Guru-In-Training** [comment]
in YAML I've put
network_to_file_image: ^2.3.4

And I'm still getting it:

```
../../../AppData/Local/Pub/Cache/hosted/pub.dartlang.org/network_to_file_image-2.3.4/lib/network_to_file_image.dart:407:7: Error: The non-abstract class '_MockHttpClientRequest' is missing implementations for these members:
 - HttpClientRequest.abort
Try to either
 - provide an implementation,
 - inherit an implementation from a superclass or mixin,
 - mark the class as abstract, or
 - provide a 'noSuchMethod' implementation.

class _MockHttpClientRequest extends HttpClientRequest {
      ^^^^^^^^^^^^^^^^^^^^^^
org-dartlang-sdk:///third_party/dart/sdk/lib/_http/http.dart:2045:8: Context: 'HttpClientRequest.abort' is defined here.
  void abort([Object? exception, StackTrace? stackTrace]);
```

**Comment by marcglasberg** [comment]
@Sam-Guru-In-Training 

So maybe this:
```
  void abort([Object exception, StackTrace stackTrace]) {}
```

Is not considered a valid override for this?

```
  void abort([Object? exception, StackTrace? stackTrace]);
```

Please see: https://github.com/marcglasberg/network_to_file_image/blob/master/lib/network_to_file_image.dart#L479

**Comment by Sam-Guru-In-Training** [comment]
OK I've edited the appropriate bit, I just fumble forward tbh.  There appears to be 2 _MockHttpClientRequest classes, in the dart file, the second class has the abort but not the first.

**Comment by marcglasberg** [comment]
@Sam-Guru-In-Training I don't get what you mean. Do you mean it's now working?

**Comment by dannycortesv** [comment]
I have tried it but still not working

**Comment by seb-almeida** [comment]
The implementation method was in _MockHttpClientResponse instead of _MockHttpClientRequest, I just moved it and it worked 👍

**Comment by marcglasberg** [comment]
@seb-almeida Thanks.

@dannycortesv @Sam-Guru-In-Training @pmrajani @abnerh69 Now please try version [2.3.5]

**Comment by seb-almeida** [comment]
2.3.5 Failed here with the very same error 😕

**Comment by davidmednikov** [comment]
Fixed in 2.3.6. Thank you!!
